internal/wal/wal_aof.go

```diff
@@ -133,13 +133,22 @@ func (wal *AOF) writeEntry(data []byte) error {
 	wal.mu.Lock()
 	defer wal.mu.Unlock()
 
+	misc := &WALEntry{}
```

Comment:
I maybe wrong but, won't changing the data to a specific `WALEntry` format here also defeat the purpose of WAL as a generic logger; which we discussed in LogCommand method signature? 
Changing to `WALEntry` format by itself is fine but having `Command` and `Args` in the WALEntry proto will limit it to dice commands only (maybe that's all we need) but I see use cases for telemetry and more where we can just tweak the current WAL to server that purpose also. 
I feel WALEntry proto should just contain metadata fields and the actual data should just be serialized into a `[]byte data` field. Thoughts @AshwinKul28 @arpitbbhayani ?